patches/minecraft/net/minecraft/block/FlowingFluidBlock.java.patch

```diff
@@ -1,6 +1,71 @@
 --- a/net/minecraft/block/FlowingFluidBlock.java
 +++ b/net/minecraft/block/FlowingFluidBlock.java
-@@ -121,13 +121,13 @@
+@@ -33,6 +33,8 @@
+    protected final FlowingFluid field_204517_c;
+    private final List<IFluidState> field_212565_c;
+ 
++   // Forge: Use the other constructor and override getFluid instead
++   @Deprecated
+    protected FlowingFluidBlock(FlowingFluid p_i49014_1_, Block.Properties p_i49014_2_) {
+       super(p_i49014_2_);
+       this.field_204517_c = p_i49014_1_;
+@@ -45,8 +47,16 @@
+ 
+       this.field_212565_c.add(p_i49014_1_.func_207207_a(8, true));
+       this.func_180632_j(this.field_176227_L.func_177621_b().func_206870_a(field_176367_b, Integer.valueOf(0)));
++      fluidStateCacheInitialized = true;
+    }
+ 
++   protected FlowingFluidBlock(Block.Properties p_i48368_1_) {
++      super(p_i48368_1_);
++      this.field_204517_c = null;
++      this.field_212565_c = Lists.newArrayList();
++      this.func_180632_j(this.field_176227_L.func_177621_b().func_206870_a(field_176367_b, Integer.valueOf(0)));
++   }
++
+    public void func_196265_a(BlockState p_196265_1_, World p_196265_2_, BlockPos p_196265_3_, Random p_196265_4_) {
+       p_196265_2_.func_204610_c(p_196265_3_).func_206891_b(p_196265_2_, p_196265_3_, p_196265_4_);
+    }
+@@ -56,17 +66,18 @@
+    }
+ 
+    public boolean func_196266_a(BlockState p_196266_1_, IBlockReader p_196266_2_, BlockPos p_196266_3_, PathType p_196266_4_) {
+-      return !this.field_204517_c.func_207185_a(FluidTags.field_206960_b);
++      return !getFluid().func_207185_a(FluidTags.field_206960_b);
+    }
+ 
+    public IFluidState func_204507_t(BlockState p_204507_1_) {
+       int i = p_204507_1_.func_177229_b(field_176367_b);
++      if (!fluidStateCacheInitialized) initFluidStateCache();
+       return this.field_212565_c.get(Math.min(i, 8));
+    }
+ 
+    @OnlyIn(Dist.CLIENT)
+    public boolean func_200122_a(BlockState p_200122_1_, BlockState p_200122_2_, Direction p_200122_3_) {
+-      return p_200122_2_.func_204520_s().func_206886_c().func_207187_a(this.field_204517_c) ? true : super.func_200124_e(p_200122_1_);
++      return p_200122_2_.func_204520_s().func_206886_c().func_207187_a(getFluid()) ? true : super.func_200124_e(p_200122_1_);
```

Comment:
while you're at it, this could be simplified to 
`p_200122_2_.func_204520_s().func_206886_c().func_207187_a(getFluid()) || super.func_200124_e(p_200122_1_);`

Reply:
It could, but I think that would also make the patch less obvious, I'd rather leave it as it is, so that people see what REALLY changed in that patch.